### PR TITLE
WIP: Added namespaces dict and namespace support in get_by_label

### PR DESCRIPTION
**Pull request by francescalb**
ontology.namespaces is now included as a dict with short name as key and iri as value.
ontology.get_by_label(label, namespace='namespace_shortname') is now supported.
same for ontology.get_by_label_all

### CODE REVIEW

File: emmo/ontology.py [hunk 1]
```diff
@@ -150,6 +151,18 @@ def __dir__(self):
         if self._dir_name:
             s.update(e.name for e in lst if hasattr(e, 'name'))
 
+        # updateing with namespace keys
+        # Attention: What to do if a prefLabel coincides with a namespace key?
+        if 'namespaces' not in self.__dict_:
```
Comment (jesper-friis) [inline emmo/ontology.py:156]: Nice. We repeat this several places. Probably a good idea to factor it out into an own method.

Comment (jesper-friis) [reply emmo/ontology.py:156]: An alternative would be to make 'namespaces' a property. Then 'namespaces' would always be there and behave as an attribute, but will under the hood be created the first time it is accessed.

Comment (francescalb) [reply emmo/ontology.py:156]: > An alternative would be to make 'namespaces' a property. Then 'namespaces' would always be there and behave as an attribute, but will under the hood be created the first time it is accessed.

NOt sure I follow.
like @property and def namespace(

**Comment by francescalb** [comment]
Most of this has been added a little bit differently, with prefix
so that all ontologies have a prefix and get_by_label accepts "prefix:label" as argument.

**Pull request closed without merging**